crates/wasmparser/src/primitives.rs

```diff
@@ -257,7 +257,7 @@ pub enum ImportSectionEntryType {
 pub struct MemoryImmediate {
     /// Alignment, stored as `n` where the actual alignment is `2^n`
     pub align: u8,
-    pub offset: u32,
+    pub offset: u64,
```

Comment:
At minimum, this field should have a comment documenting memory32 vs memory64 and the guarantee that you'll only get offsets that fit in a `u32` when 64 bit memories aren't enabled.